.github/workflows/ci.yaml

```diff
@@ -0,0 +1,80 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+    - track/**
+
+# kill all previous jobs in the same branch if new commits are pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  quality-checks:
+    name: Quality Checks
+    strategy:
+      matrix:
+        charm:
+          - knative-eventing
+          - knative-operator
+          - knative-serving
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/_quality-checks.yaml@all
+    with:
+      charm-path: ./charms/${{ matrix.charm }}
+      charm-name: ${{ matrix.charm }}
+
+  unit-tests:
+    name: Unit Tests
+    strategy:
+      matrix:
+        charm:
+          - knative-eventing
+          - knative-operator
+          - knative-serving
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/_unit-tests.yaml@all
+    secrets: inherit
+    with:
+      charm-name: ${{ matrix.charm }}
+
+# uncomment when integration tests are added
```

Comment:
This is something interesting to discuss and it has to do with the way we write tests. In multi-charm repositories, we have the option to add individual integration tests, if you compare kfp-operators and knative-operators you'll find that we don't have a pattern, these tests may be present or not. We should define the strategy and make a standard, then we'd be able to uncomment or remove this section entirely depending on our decision.